tiledb/sm/subarray/test/unit_range_subset.cc

```diff
@@ -323,67 +340,322 @@ TEMPLATE_TEST_CASE_SIG(
+    for (auto range : ranges) {
+      range_subset.add_range(range);
+    }
+    CHECK(range_subset.num_ranges() == 4);
+
+    // Sort and merge ranges.
+    ThreadPool pool{2};
+    range_subset.sort_and_merge_ranges(&pool, merge);
+
+    // Check range results.
+    if (merge) {
+      CHECK(range_subset.num_ranges() == 1);
+      check_subset_range_values(range_subset, 0, data1[0], data2[1]);
+    } else {
+      CHECK(range_subset.num_ranges() == 4);
+      check_subset_range_values(range_subset, 0, data1[0], data1[1]);
+      check_subset_range_values(range_subset, 1, data4[0], data4[1]);
+      check_subset_range_values(range_subset, 2, data3[0], data3[1]);
+      check_subset_range_values(range_subset, 3, data2[0], data2[1]);
+    }
+  }
+
+  SECTION("Partially overlapping") {
+    // Add ranges.
+    T data1[2] = {0, 2};
+    T data2[2] = {1, 4};
+    T data3[2] = {7, 9};
+    T data4[2] = {3, 5};
+    std::vector<Range> ranges = {
+        Range(data1, 2 * sizeof(T)),
+        Range(data2, 2 * sizeof(T)),
+        Range(data3, 2 * sizeof(T)),
+        Range(data4, 2 * sizeof(T))};
+    for (auto range : ranges) {
+      range_subset.add_range(range);
+    }
+    CHECK(range_subset.num_ranges() == 4);
+
+    // Sort and merge ranges.
+    ThreadPool pool{2};
+    range_subset.sort_and_merge_ranges(&pool, merge);
+
+    // Check range results.
+    if (merge) {
+      CHECK(range_subset.num_ranges() == 2);
+      check_subset_range_values(range_subset, 0, data1[0], data4[1]);
+      check_subset_range_values(range_subset, 1, data3[0], data3[1]);
+    } else {
+      CHECK(range_subset.num_ranges() == 4);
+      check_subset_range_values(range_subset, 0, data1[0], data1[1]);
+      check_subset_range_values(range_subset, 1, data2[0], data2[1]);
+      check_subset_range_values(range_subset, 2, data4[0], data4[1]);
+      check_subset_range_values(range_subset, 3, data3[0], data3[1]);
+    }
   }
 }
 
-TEST_CASE("RangeSetAndSuperset::sort - STRING_ASCII") {
+TEST_CASE(
+    "RangeSetAndSuperset::sort_and_merge_ranges - STRING_ASCII",
+    "[sort_and_merge_ranges][string]") {
+  bool merge = GENERATE(true, false);
   Range range{};
   RangeSetAndSuperset range_subset{Datatype::STRING_ASCII, range, false, false};
-  SECTION("Sort 2 reverse ordered non-overlapping ranges") {
-    // Set ranges.
-    const std::string d1{"cat"};
-    const std::string d2{"dog"};
-    const std::string d3{"ax"};
-    const std::string d4{"bird"};
-    Range r1{d1, d2};
-    range_subset.add_range(r1);
-    Range r2{d3, d4};
-    range_subset.add_range(r2);
+
+  SECTION("Empty ranges") {
+    CHECK(range_subset.num_ranges() == 0);
+
+    // Try to sort and merge ranges.
+    ThreadPool pool{2};
+    range_subset.sort_and_merge_ranges(&pool, merge);
+
+    // Check range results.
+    CHECK(range_subset.num_ranges() == 0);
+  }
+
+  SECTION("Adjacent, sorted ranges") {
+    // Note: only char ranges will coalesce, string ranges will not
```

Comment:
Char ranges should only be for DataType::CHAR.

Reply:
we don't allow char dimensions anyway so perhaps the char logic should be removed regardless...?